linkerd/proxy/server-policy/src/lib.rs

```diff
@@ -14,10 +16,11 @@ pub use self::{
 };
 pub use linkerd_http_route as route;
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug)]
 pub struct ServerPolicy {
     pub protocol: Protocol,
     pub meta: Arc<Meta>,
+    pub local_rate_limit: Arc<HttpLocalRateLimit>,
```

Comment:
This isn't a big problem, but it seems like it would be most consistent (with both the proxy prior art and API) for the rate limit to be encoded in the `Protocol`, since it's not relevant for certain protocol variants.

```rust
    Detect {
        http: Arc<[http::Route]>,
        http_local_rate_limit: Arc<LocalRateLimit>
        timeout: time::Duration,
        tcp_authorizations: Arc<[Authorization]>,
    },
    Http1(Arc<[http::Route]>, Arc<LocalRateLimit>),
    Http2(Arc<[http::Route]>, Arc<LocalRateLimit>),
```

Reply:
The problem is that governor's `RateLimiter` doesn't implement `PartialEq`, which `Protocol` requires. That requirement can be lifted from the types depending on `Protocol`, except for `ServerPermit`, which uses it for test assertions. So we could either manually implement `PartialEq` for `ServerPermit` just so that those tests aren't affected, or implement it for our `LocalRateLimit` wrapper you suggested above but it'd only check for equality on rps and ids. Any thoughts?

Reply:
After looking at your follow-on PR, this change probably isn't worth it -- it's more complicated to consume the rate limit without much benefit.